src/rez/utils/system.py

```diff
@@ -36,3 +37,18 @@ def popen(args, **kwargs):
             kwargs["stdin"] = subprocess.PIPE
 
     return subprocess.Popen(args, **kwargs)
+
+
+@contextmanager
+def change_dir(dir_path):
```

Comment:
This is very similar to existing `rez.utils.filesystem.retain_cwd`. I would update this existing func to be able to chdir to a specified path, and use that instead.